integrations/langchain/tests/unit_tests/test_vector_search_retriever_tool.py

```diff
@@ -0,0 +1,93 @@
+from typing import Any, Dict, List, Optional
+
+import pytest
+from langchain_core.embeddings import Embeddings
+from langchain_core.tools import BaseTool
+
+from databricks_langchain import ChatDatabricks, VectorSearchRetrieverTool
+from tests.utils.chat_models import llm, mock_client  # noqa: F401
+from tests.utils.vector_search import (  # noqa: F401
+    ALL_INDEX_NAMES,
+    DELTA_SYNC_INDEX,
+    EMBEDDING_MODEL,
+    mock_vs_client,
+    mock_workspace_client,
+)
+
+
+def init_vector_search_tool(
+    index_name: str,
+    columns: Optional[List[str]] = None,
+    tool_name: Optional[str] = None,
+    tool_description: Optional[str] = None,
+    embedding: Optional[Embeddings] = None,
+    text_column: Optional[str] = None,
+    endpoint: Optional[str] = None,
+) -> VectorSearchRetrieverTool:
+    kwargs: Dict[str, Any] = {
+        "index_name": index_name,
+        "columns": columns,
+        "tool_name": tool_name,
+        "tool_description": tool_description,
+        "embedding": embedding,
+        "text_column": text_column,
+        "endpoint": endpoint,
+    }
+    if index_name != DELTA_SYNC_INDEX:
+        kwargs.update(
+            {
+                "embedding": EMBEDDING_MODEL,
+                "text_column": "text",
+            }
+        )
+    return VectorSearchRetrieverTool(**kwargs)  # type: ignore[arg-type]
+
+
+@pytest.mark.parametrize("index_name", ALL_INDEX_NAMES)
+def test_init(index_name: str) -> None:
+    vector_search_tool = init_vector_search_tool(index_name)
+    assert isinstance(vector_search_tool, BaseTool)
+
+
+@pytest.mark.parametrize("index_name", ALL_INDEX_NAMES)
+def test_chat_model_bind_tools(llm: ChatDatabricks, index_name: str) -> None:
+    from langchain_core.messages import AIMessage
+
+    vector_search_tool = init_vector_search_tool(index_name)
+    llm_with_tools = llm.bind_tools([vector_search_tool])
+    response = llm_with_tools.invoke("Which city is hotter today and which is bigger: LA or NY?")
+    assert isinstance(response, AIMessage)
+
+
+@pytest.mark.parametrize("index_name", ALL_INDEX_NAMES)
+@pytest.mark.parametrize("columns", [None, ["id", "text"]])
+@pytest.mark.parametrize("tool_name", [None, "test_tool"])
+@pytest.mark.parametrize("tool_description", [None, "Test tool for vector search"])
+@pytest.mark.parametrize("embedding", [None, EMBEDDING_MODEL])
+@pytest.mark.parametrize("text_column", [None, "text"])
+@pytest.mark.parametrize("endpoint", [None, "test_endpoint"])
+def test_vector_search_retriever_tool_combinations(
+    index_name: str,
+    columns: Optional[List[str]],
+    tool_name: Optional[str],
+    tool_description: Optional[str],
+    embedding: Optional[Any],
+    text_column: Optional[str],
+    endpoint: Optional[str],
```

Comment:
We shouldn't need the `endpoint` argument anymore right?

Reply:
Good catch updated!